File: Assignment_4/java/flyweight/src/Text.java
public class Text {
    private String text;
    TextType textType = null;

    Text(String text) {
        SetText(text);
    }

    Text(String text, TextType text_type) {
        SetText(text);
        SetTextType(text_type);
    }

    void SetText(String text) {
        this.text = text;
    }

    void SetTextType(TextType text_type) {
        textType = text_type;
    }

    String Gettext() {
        return text;
    }

    TextType GetTextType() {
        return textType;
    }
}

File: Assignment_4/cpp/flyweight/Text.cpp
//
// Created by Sean on 4/21/2023.
//

#include "Text.h"
#include "TextType.h"

Text::Text(std::string text)
{
    SetText(std::move(text));
}

Text::Text(std::string text, TextType * text_type)
{
  SetText(std::move(text));
  SetTextType(text_type);
}

void Text::SetText(std::string new_text)
{
  text = std::move(new_text);
}

void Text::SetTextType(TextType * text_type)
{
  textType = text_type;
}

const std::string & Text::GetText()
{
  return text;
}
